chapter/related-work.md

```diff
@@ -8,9 +8,58 @@ Additionally, this part aims to recognize alternative approaches to the LSP, in
+This way it is able to fully concentrate on source analysis instead of integration with specific editors by building custom GUI elements and being restricted to editors extension interface.
+
+At the time of writing the LSP is available in version `3.16` [@Spec].
+Microsoft's official website lists 172 implementations of the LSP[@implementations] for an equally impressive number of languages.
+
+Assessing any majority of these is outside the scope of this work.
+Yet, a few implementations stand out due to their sophisticate architecture, features, popularity or closeness to the presented work.
+
 ### Considerable dimensions
 
-#### Language Complexity
+To be able to compare and describe each project objectively and comprehensively, the focus will be on the following dimensions.
+
+Target Language
+  ~ The complexity of implementing language servers is influenced severely by the targeted language.
+    Feature rich languages naturally require more sophisticated solutions than simpler ones.
+    Conversely, self-hosted and interpreted languages can often leverage existing tooling.
+
+Features
+  ~ The LSP defines an extensive array of capabilities.
+    The implementation of these protocol features is optional and servers and clients are able to communicate a set of *mutually supported* capabilities.
+  ~ The Langserver.org^[https://langserver.org] project identified six basic capabilities that are most widely supported:
+
+    1. `Code completion`
+    2. `Hover`,
+    3. `Jump to def`,
+    4. `Workspace symbols`,
+    5. `Find references`
+    6. Diagnostics
+
+  ~ Yet, not all of these are applicable in every case and some LSP implementations reach for a much more complete coverage of the protocol.
+
+File Processing
+  ~ Most language servers use very different methods of handling the source code they analyze.
+    The means are mainly influenced by the complexity of the language.
+    Distinctions appear in the way servers process *file indexes and changes* and how they respond to *requests*.
+  ~ The LSP supports sending updates in form of diffs of atomic changes and complete transmission of changed files.
+    The former requires methods of incremental parsing and analysis which are difficult by itself, but are able to process files much more quickly.
+    An incremental approach makes use of an internal representation of the source code that allows to quickly derive analytic results from and can be updated efficiently.
+    Additionally, to facilitate the parsing, it must be able to provide a parser with the right context to correctly parse a changed fragment of code.
+    On the contrary, most language servers process file changes by re-indexing the entire file, updating their internal model much more broadly.
+    This is a more approachable method.
+    Yet, it is less performant since entire files need to be processed, which becomes more noticeable as file sizes and edit frequency increase.
```

Comment:
I think you can emphasize much more the aspect that incrementality is about *small changes imply small reprocessing* in the paragraph above. For example, entire files also need to be processed by incremental LSP, but the point is that they don't need to be entirely REprocessed upon small changes.